cmd/wallet/deposits.go

```diff
@@ -0,0 +1,189 @@
+		})
+	}
+
+	// Print
+	if err := printTable(deposits); err != nil {
+		return fmt.Errorf("failed to print deposits table: %v", err)
+	}
+
+	return nil
+}
+
+func removeSwept(deposits []depositEntry) []depositEntry {
+	result := []depositEntry{}
+	for _, deposit := range deposits {
+		if deposit.isSwept {
+			continue
+		}
+		result = append(result, deposit)
+	}
+	return result
+}
+
+func getDeposits(tbtcChain tbtc.Chain, walletStr string) ([]depositEntry, error) {
+	logger.Infof("reading deposits from chain...")
+
+	filter := &tbtc.DepositRevealedEventFilter{}
+	if len(walletStr) > 0 {
+		walletPublicKeyHash, err := hexToWalletPublicKeyHash(walletStr)
+		if err != nil {
+			return []depositEntry{}, fmt.Errorf("failed to extract wallet public key hash: %v", err)
+		}
+
+		filter.WalletPublicKeyHash = [][20]byte{walletPublicKeyHash}
+	}
+
+	depositRevealedEvent, err := tbtcChain.PastDepositRevealedEvents(filter)
+	if err != nil {
+		return []depositEntry{}, fmt.Errorf(
+			"failed to get past deposit revealed events: [%w]",
+			err,
+		)
+	}
+
+	logger.Infof("found %d DepositRevealed events", len(depositRevealedEvent))
+
+	result := make([]depositEntry, len(depositRevealedEvent))
+	for i, event := range depositRevealedEvent {
+		logger.Debugf("getting details of deposit %d/%d", i+1, len(depositRevealedEvent))
+
+		depositKey := buildDepositKey(event.FundingTxHash, event.FundingOutputIndex)
+
+		depositRequest, err := tbtcChain.GetDepositRequest(event.FundingTxHash, event.FundingOutputIndex)
+		if err != nil {
+			return result, fmt.Errorf(
+				"failed to get deposit request: [%w]",
+				err,
+			)
+		}
+
+		result[i] = depositEntry{
+			wallet:                        event.WalletPublicKeyHash,
+			depositKey:                    depositKey,
+			revealedAtBlock:               event.BlockNumber,
+			isSwept:                       depositRequest.SweptAt.After(depositRequest.RevealedAt),
+			fundingTransactionHash:        event.FundingTxHash,
+			fundingTransactionOutputIndex: event.FundingOutputIndex,
+			amountBtc:                     convertSatToBtc(float64(depositRequest.Amount)),
+		}
+	}
+
+	return result, nil
+}
+
+func printTable(deposits []depositEntry) error {
+	w := tabwriter.NewWriter(os.Stdout, 2, 4, 1, ' ', tabwriter.AlignRight)
+	fmt.Fprintf(w, "index\twallet\tvalue (BTC)\tdeposit key\tfunding transaction\tswept\t\n")
+
+	for i, deposit := range deposits {
+		fmt.Fprintf(w, "%d\t%s\t%.5f\t%s\t%s\t%v\t\n",
+			i,
+			"0x"+hex.EncodeToString(deposit.wallet[:]),
+			deposit.amountBtc,
+			deposit.depositKey,
+			fmt.Sprintf(
+				"%s:%d",
+				deposit.fundingTransactionHash.Hex(bitcoin.ReversedByteOrder),
+				deposit.fundingTransactionOutputIndex,
+			),
+			deposit.isSwept,
+		)
+	}
+
+	if err := w.Flush(); err != nil {
+		return fmt.Errorf("failed to flush the writer: %v", err)
+	}
+
+	return nil
+}
+
+func hexToWalletPublicKeyHash(str string) ([20]byte, error) {
+	walletHex, err := hexutil.Decode(str)
```

Comment:
I don't think we should use `github.com/ethereum/go-ethereum` package here as it makes the Ethereum-specific implementation details leak into a code that should be chain-agnostic. The only place this package can be used safely is `pkg/chain/ethereum`. Hence, I propose using `hex.DecodeString` instead. Just remember about stripping the `0x` in case it exists at the beginning.

Reply:
The reason why I used `hexutil` is that it handles prefixes. I'll refactor it to not relay on the `github.com/ethereum/go-ethereum`.

Reply:
3f4e2b3f01d8adf52ea32633485e043c240692ce